server/driver/wivrn_session.cpp

```diff
@@ -418,8 +424,18 @@ void wivrn_session::operator()(from_headset::trackings && tracking)
 		        std::string(magic_enum::enum_name(roles.right_profile)).c_str(),
 		        std::string(magic_enum::enum_name(right)).c_str());
 		std::lock_guard lock(roles_mutex);
+		if (left == XRT_DEVICE_EXT_HAND_INTERACTION)
+			roles.left = left_hand_interaction_index;
+		else
+			roles.left = left_hand_index;
```

Comment:
We should only assign to `roles.left` if the previous role is one of the WiVRn managed ones.
In case of lighthouse devices, we could have the left hand on WiVRn, while the right one is lighthouse.

In that case, only the `roles.left` shall be updated, even if both interaction profiles changed.
The `XRT_DEVICE_INVALID` profile can be returned by the headset if there is no active interaction profile, so this is not a sufficient check either.